### PR TITLE
Python builtin methods iter

### PR DESCRIPTION
**Pull request by Varun2711**
### Description

Added a new term entry for python built-in function iter()
The entry can be found in the markdown file: `iter.md` on the path: `content\python\concepts\built-in-functions\terms\iter\iter.md` 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by cigar-galaxy82** [comment]
Hi, @Varun2711 I see that there is a change in yarm.lock file please revert those changes.

**Comment by Varun2711** [comment]
Hi @cigar-galaxy82, I have reverted the changes you requested

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+```pseudo
+iter(object)
+```
+
+```pseudo
+iter(object, sentinel)
+```
+
+- `object` parameter supplies the object on which the iter() method will return the iterator. The type of this parameter depends on the presence of the optional second `sentinel` parameter. In the case where the sentinel parameter is not given, the object parameter type must be one that supports iteration or a sequence. These types include: list, str, and tuple among others. You may also pass in objects that you've defined with either **\_\_iter\_\_()** or **\_\_getitem\_\_()** methods. When the sentinel parameter is present, the object parameter must be callable.
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:26]:
```suggestion
- `object`: A required argument represents an object such as a list, tuple, string, or any object that follows iterator protocol.
```

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+- `sentinel` is the parameter that stores the value which represents the end of the sequence. When the callable object returns this value, the iterator will terminate.
```
[the remark this answers is not in the document]
Comment (cigar-galaxy82) [reply content/python/concepts/built-in-functions/terms/iter/iter.md:28]: After this as a note mentioned when sentinel parameter is used like in cases used with custom iterators

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## example one iter() with no sentinel parameter
+fruits = ['apples', 'bananas', 'oranges']
+
+my_iterator = iter(fruits)
+print(my_iterator)
+
+## example two iter() with a sentinel parameter
+func_iterator = iter(partial(random.randint, 1, 10), 7)
+print(func_iterator)
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:44]:
```suggestion
from functools import partial
import random

# iter() with no sentinel parameter
fruits = ['apples', 'bananas', 'oranges']
my_iterator = iter(fruits)
print(next(my_iterator))
print(next(my_iterator))
print(next(my_iterator))
```

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+```shell
+<list_iterator object at 0x000001CCFDCAA470>
+<callable_iterator object at 0x000001CCFDE37910>
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:49]:
```suggestion
apples
bananas
oranges
```

Comment (cigar-galaxy82) [reply content/python/concepts/built-in-functions/terms/iter/iter.md:49]: Add another example that show use of iter with both argument

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Codebyte Example (if applicable)
+
+```codebyte/python
+technologies = ['next.js', 'react', 'node.js', 'postgreSql', 'firebase', 'tailwindcss']
+
+tech_iterator = iter(technologies)
+
+for tech in tech_iterator:
+  print(tech)
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:60]:
```suggestion
from functools import partial

technologies = ['next.js', 'react', 'node.js', 'postgreSql', 'firebase', 'tailwindcss']
tech_iterator = iter(technologies)
for tech in tech_iterator:
  print(tech)
```

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+func_iterator = iter(partial(random.randint, 1, 10), 7)
+print(func_iterator)
+```
+
+```shell
+<list_iterator object at 0x000001CCFDCAA470>
+<callable_iterator object at 0x000001CCFDE37910>
+```
+
+## Codebyte Example (if applicable)
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:52]:
```suggestion
## Codebyte Example
```

File: content/python/concepts/built-in-functions/terms/iter/iter.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import randint from random
+
+def callback_fn():
+  return randint(1,10)
+
+call_iterator = iter(callback_fn, 7)
+
+for val in call_iterator:
+  print(val)
+```
```
Comment (cigar-galaxy82) [inline content/python/concepts/built-in-functions/terms/iter/iter.md:73]: Add them in a single codebyte block

File: content/python/concepts/built-in-functions/terms/iter/iter.md (hunk 3 above)
Comment (cigar-galaxy82) [reply content/python/concepts/built-in-functions/terms/iter/iter.md:44]: The reason for this is we want to show a simple example on how to use iter and current example show the location which can be confusing

**Comment by cigar-galaxy82** [comment]
Hi @Varun2711 thank you for contributing to Codecademy docs, I have suggested a few changes please modify.

**Comment by Varun2711** [comment]
Hello @cigar-galaxy82, I updated the file with the changes you suggested. Please look over them and let me know if there are any improvements to be made. Thanks!

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing @Varun2711 😄 

The entry is ready to be merged! 🚀

**Pull request merged**